packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -66,49 +66,71 @@ const getProvider = async (state?: Partial<MetadataProviderCredentials>) => {
     return providerInstance;
 };
 
-export const enableMetadata = () => (dispatch: Dispatch) => {
-    dispatch({
-        type: METADATA.ENABLE,
-    });
-};
-
-export const disposeMetadata = () => (dispatch: Dispatch, getState: GetState) => {
+/**
+ * dispose metadata from all labelable objects.
+ * Has keys parameter,
+ * if true - dispose also metadata keys
+ * if false - dispose only metadata values
+ */
+export const disposeMetadata = (keys?: boolean) => (dispatch: Dispatch, getState: GetState) => {
     getState().wallet.accounts.forEach(account => {
+        const updatedMetadata = { ...account.metadata };
+
+        // always remove metadata  values
+        updatedMetadata.outputLabels = {};
+        updatedMetadata.addressLabels = {};
+        updatedMetadata.accountLabel = '';
```

Comment:
Could be simplified:
```
const updatedMetadata = {
    ...account.metadata,
    // always remove metadata  values
    outputLabels: {},
    addressLabels: {},
    accountLabel: '',
};
```

Reply:
If you wish.. But I don't think it is easier to read. Consider
After your suggestion: 

```javascript
const updatedMetadata = {
  ...account.metadata,
  // always remove metadata  values
  outputLabels: {},
  addressLabels: {},
  accountLabel: '',
};

// and sometimes remove also keys (information we can only if device is connected)
if (keys) {
  updatedMetadata.fileName = '';
  updatedMetadata.aesKey = '';
}

```

My original idea why to write it the way I did was to express 2 stages of deleting values in 2 individual steps. 

```javascript
const updatedMetadata = { ...account.metadata };

// always remove metadata  values
updatedMetadata.outputLabels = {};
updatedMetadata.addressLabels = {};
updatedMetadata.accountLabel = '';

// and sometimes remove also keys (information we can only if device is connected)
if (keys) {
  updatedMetadata.fileName = '';
  updatedMetadata.aesKey = '';
}

```

Reply:
I see it more as the first step being run in any case, it's clearer to initialize the object like we want it to be rather than reassigning values to its properties right below. Both work, so I'm nitpicking here again :slightly_smiling_face:

Reply:
I did as you wish but with my heart bleeding :dagger: :heart: